doc/examples/filters/plot_max_tree.py

```diff
@@ -0,0 +1,168 @@
+for a large family of morphological filters.
+
+If we apply a threshold operation to an image, we obtain a binary image
+containing a certain number of connected components. If we apply a lower
+threshold, we observe that the connected components we obtain now contain
+all connected components obtained by the higher threshold. With this, we can
+define a graph representation of the components: whenever a connected component
+A obtained by thresholding with threshold t1 is contained in a component B
+obtained by thresholding with threshold t1 < t2, we say that B is the parent
+of A. The resulting tree structure is called a component tree. The max-tree
+is a compact representation of such a component tree.
+
+In this example we give an intuition what a max-tree is.
+
+References
+----------
+.. [1] Salembier, P., Oliveras, A., & Garrido, L. (1998). Antiextensive
+       Connected Operators for Image and Sequence Processing.
+       IEEE Transactions on Image Processing, 7(4), 555-570.
+.. [2] Berger, C., Geraud, T., Levillain, R., Widynski, N., Baillard, A.,
+       Bertin, E. (2007). Effective Component Tree Computation with
+       Application to Pattern Recognition in Astronomical Imaging.
+       In International Conference on Image Processing (ICIP) (pp. 41-44).
+.. [3] Najman, L., & Couprie, M. (2006). Building the component tree in
+       quasi-linear time. IEEE Transactions on Image Processing, 15(11),
+       3531-3539.
+.. [4] Carlinet, E., & Geraud, T. (2014). A Comparative Review of
+       Component Tree Computation Algorithms. IEEE Transactions on Image
+       Processing, 23(9), 3885-3895.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.lines import Line2D
+from skimage.morphology import build_max_tree
+import numpy.random as rd
+import pdb
+
+
+def plot_img(image, ax, title, plot_text,
+             image_values):
+    ax.imshow(image, cmap='gray', aspect='equal', vmin=0, vmax=np.max(image))
+    ax.axis('off')
+    ax.set_title(title)
+
+    for x in np.arange(-0.5, image.shape[0], 1.0):
+        ax.add_artist(Line2D((x, x), (-0.5, image.shape[0]-0.5),
+                             color='blue', linewidth=2))
+
+    for y in np.arange(-0.5, image.shape[1], 1.0):
+        ax.add_artist(Line2D((-0.5, image.shape[1]), (y, y),
+                             color='blue', linewidth=2))
+
+    if plot_text:
+        k = 0
+        for i in range(image_values.shape[0]):
+            for j in range(image_values.shape[1]):
+                ax.text(j, i, image_values[i, j], fontsize=8,
+                        horizontalalignment='center',
+                        verticalalignment='center',
+                        color='red')
+                k += 1
+    return
+
+# small example image
+image = np.array([[5, 4, 4, 3, 3],
+                  [5, 8, 8, 0, 0],
+                  [2, 1, 2, 2, 1],
+                  [4, 4, 4, 7, 5],
+                  [5, 2, 3, 6, 6]], dtype=np.uint8)
+width, height = image.shape
```

Comment:
Initially, I wanted to show an image containing also plateaus because that illustrates the canonical pixels. But perhaps you are right ... I will try this. 

Two other options: 
- I could just show this for a 1-D signal. 
- I could plot the max-tree as a graph. This is certainly the best option. Do you have an idea which graph plotting library I could use for this? I think this would be clearer. 

Other than that, I will just try out your suggestion.

Reply:
@ThomasWalter for the first option (no canonization, very simple tree), if you get the tree into a networkx graph, I've already written up a function to overlay a networkx graph on an image [here](https://github.com/jni/skan/blob/54531a9ad646a73b0a31f713d035aff3243ebe90/skan/draw.py#L301-L339) (BSD licensed, so you can just copy whatever you need.)

1D seems boring. =P

Showing only the graph, I think, would be confusing. It would be hard for users to figure out the correspondence between the graph and the image.

Another (potentially much more laborious) option is to show the graph with plateaus both pre- and post-canonization. I presume you have better things to do than work on this, but I think an animation showing this might have very high pedagogical value.

For me, the canonization is secondary to the concept of the max-tree (it's a tweak, rather than the main idea), so I vote for a simple 2D example that doesn't require canonization.

@stefanv Grave might be the best for simple graph plotting, but that package might be too immature even for our docs.

Reply:
OK, I tested. The problem is that the graph is still too messy to be useful. 

@jni : thanks for your code; that looks very nice. However, I think that here the problem is really that the max-tree remains messy in any case if we plot this on top of the image. I think the best would be to plot the graph structure. This is what people do in the papers I cite ...

Reply:
Fair enough. You could plot the tree next to an image where each pixel is numbered, so people can make the correspondence between graph nodes and pixel sets. Does that make sense?

Reply:
Yes, this is what I intended to do. However, this week will be very challenging for me, as I will not be in the lab :-(

Reply:
Just to comment on this again; I have now moved the `plot_max_tree.py` to `yy_dev` in the `example` folder. This is really something for developers, rather than general users who will not be so interested in understanding the max-tree itself. It is now accompanied by text that explains what we actually see.